src/VirtualClient/VirtualClient.Actions/Kafka/KafkaExecutor.cs

```diff
@@ -0,0 +1,209 @@
+﻿namespace VirtualClient
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Net;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Newtonsoft.Json;
+    using VirtualClient.Actions.Kafka;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// The Kafka client action
+    /// </summary>
+    [UnixCompatible]
+    [WindowsCompatible]
+    public class KafkaExecutor : VirtualClientComponent
```

Comment:
make it abstract if not intended for direct use and call it like KafkaBaseExecutor

Reply:
Yeah.